### PR TITLE
Add missing `scanner()` method for `re.Pattern` objects

### PR DESCRIPTION
**Pull request by AlexWaygood**
This one's a bit weird; feedback welcome!

### CODE REVIEW

File: stdlib/typing.pyi [hunk 1]
```diff
@@ -597,12 +597,23 @@ class Match(Generic[AnyStr]):
     if sys.version_info >= (3, 9):
         def __class_getitem__(cls, item: Any) -> GenericAlias: ...
 
+# Not actually defined in typing (or re!) - is _sre.SRE_Scanner at runtime
+# Also not documented anywhere
+@_final
+class _SRE_Scanner(Generic[AnyStr]):
+    pattern: Pattern[AnyStr]
+    # Cannot be directly instantiated
+    def __new__(cls, *args: Any, **kwargs: Any) -> NoReturn: ...  # type: ignore
```
Comment (Akuli) [inline stdlib/typing.pyi:606]: Why does this need `# type: ignore`?

**Review by Akuli — COMMENTED** [review]
This one is a bit weird. On the one hand, stubs should match the implementation closely. But on the other hand, we probably don't want `.scanner()` to show up in autocompletions, so that people would just call `.match()` and `.search()` directly, or `functools.partial(foo.match, string)` instead of `foo.scanner(string).match` if a callback is needed. I would like to know what other maintainers think about this.

File: stdlib/typing.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stdlib/typing.pyi:606]: Because `__new__` has to return an instance of `cls`, according to normal rules.

**Comment by JelleZijlstra** [comment]
What is the motivation for this PR? The method seems undocumented, so our policy is to add it if a user who is using it asks for it, but I don't think we should be adding undocumented methods just for completeness' sake.

**Comment by AlexWaygood** [comment]
> I don't think we should be adding undocumented methods just for completeness' sake.

Point taken 🙂

**Comment by AlexWaygood** [comment]
> I don't think we should be adding undocumented methods just for completeness' sake.

Point taken 🙂

**Pull request closed without merging**